src/main/java/net/minecraftforge/common/util/Constants.java

```diff
@@ -49,4 +49,132 @@
+        public static final int ANVIL_USE_SOUND                 = 1030;
+        public static final int ANVIL_LAND_SOUND                = 1031;
+        public static final int PORTAL_TRAVEL_SOUND             = 1032;
+        public static final int CHORUS_FLOWER_GROW_SOUND        = 1033;
+        public static final int CHORUS_FLOWER_DEATH_SOUND       = 1034;
+        public static final int BREWING_STAND_BREW_SOUND        = 1035;
+        public static final int IRON_TRAPDOOR_CLOSE_SOUND       = 1036;
+        public static final int IRON_TRAPDOOR_OPEN_SOUND        = 1037;
+        /**
+         * data is the direction of the smoke, as a grid around the dispenser as follows (with 4 being the position of the dispenser):
+         *        N
+         *    0 | 1 | 2
+         *    ----------
+         * W  3 | 4 | 5  E
+         *    ----------
+         *    6 | 7 | 8
+         *        S
+         * Setting the data to 4 will mean the particles won't spawn
+         */
+        public static final int DISPENSER_SMOKE                 = 2000;
+
+        /**
+         * data is the state id of the block broken {@link net.minecraft.block.Block#getStateId}
+         */
+        public static final int BREAK_BLOCK_EFFECTS             = 2001;
+        /**
+         * data is the rgb color int that should be used for the potion particles
+         */
+        public static final int SPLASH_POTION_EFFECT            = 2002;
+        public static final int ENDER_EYE_SHATTER               = 2003;
+        public static final int MOB_SPAWNER_PARTICLES           = 2004;
+        /**
+         * data is the amount of particles to spawn. If data is 0 then there will be 15 particles spawned
+         */
+        public static final int BONEMEAL_PARTICLES              = 2005;
+        public static final int DRAGON_FIREBALL_HIT             = 2006;
+        /**
+         * data is the rgb color int that should be used for the potion particles
+         */
+        public static final int LINGERING_POTION_EFFECT         = 2007;
+        public static final int GATEWAY_SPAWN_EFFECTS           = 3000;
+        public static final int ENDERMAN_GROWL_SOUND            = 3001;
+    }
+
+
+    /**
+     * The flags used when calling {@link net.minecraft.world.World#setBlockState}
+     * Can be found from World#markAndNotifyBlock and RenderGlobal#notifyBlockUpdate
+     * Flags can be OR-ed
+     */
+    public static class BlockFlags {
+        /**
+         * Calls neighborChanged on surrounding blocks
+         */
+        public static final int NOTIFY_NEIGHBORS     = 0b00001;
+        /**
+         * Sends the update to the client
+         */
+        public static final int SEND_TO_CLIENTS      = 0b00010;
+        /**
+         * Stops the blocks from being marked for a render update
+         */
+        public static final int NO_RERENDER          = 0b00100;
+        /**
+         * Makes the block be re-rendered immediately, on the main thread
```

Comment:
add or clarify that if NO_RERENDER is set then this flag doesn't have any effect